client-for-auth-ii/src/components/Register.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { StyledForm } from '../styles/StyledForm';
+
+const initalUser = {
+  username: '',
+  password: '',
+  department: ''
+};
+
+const url = process.env.REACT_APP_API_URL;
+
+class Register extends Component {
+  state = {
+    user: { ...initalUser },
+    message: ''
+  };
+
+  handleInputChange = e => {
+    const { name, value } = e.target;
+    this.setState({ user: { ...this.state.user, [name]: value } });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    axios
+      .post(`${url}/api/register`, this.state.user)
+      .then(res => {
+        if (res.status === 201) {
+          this.setState({
+            message: 'Register Successful',
+            user: { ...initalUser }
+          });
+          this.props.history.push('/login');
+        }
+      })
+      .catch(err => {
+        this.setState({
+          message: err.response.data.message,
+          user: { ...initalUser }
+        });
+      });
+  };
+
+  render() {
+    const { username, password, department } = this.state.user;
+    return (
+      <div>
+        <StyledForm onSubmit={this.handleSubmit}>
+          <h3>Sign Up</h3>
+          <input
```

Comment:
It's considered good practice to use labels for accessibility reasons, rather than relying on placeholder to inform which input is which